cluster/images/etcd-version-monitor/etcd-version-monitor.go

```
@@ -0,0 +1,275 @@
+		},
+		[]string{"backend_version", "unit", "etcdctl_version"})
+	etcdGRPCRequestsTotal	= prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name: "grpc_requests_total",
+			Help: "Counter of received grpc requests, labeled by grpc method and grpc service names",
+		},
+		[]string{"grpc_method", "grpc_service"})
+)
+
+// Struct for unmarshalling the json response from etcd's /version endpoint.
+type EtcdVersion struct {
+	ServerVersion		string `json:"etcdserver"`
+	ClusterVersion		string `json:"etcdcluster"`
+}
+
+// Function for periodically fetching etcd version info.
+func getVersion() {
+	// Create the get request for the etcd endpoint.
+	req, err := http.NewRequest("GET", etcdVersionScrapeURI, nil)
+	if err != nil {
+		log.Fatal("NewRequest: ", err)
+		return
+	}
+
+	for {
+		// Send the get request and receive a response.
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			log.Fatal("Do: ", err)
+			time.Sleep(scrapeTimeout)
+			continue
+		}
+
+		// Obtain EtcdVersion from the JSON response.
+		var version EtcdVersion
+		if err := json.NewDecoder(resp.Body).Decode(&version); err != nil {
+			log.Fatal("JSON decoder for etcd version: ", err)
+			time.Sleep(scrapeTimeout)
+			continue
+		}
+		resp.Body.Close()
+
+		// Record the info in the prometheus metric through labels.
+		etcdVersionFetchCount.With(prometheus.Labels{
+			"server_version": version.ServerVersion,
+			"cluster_version": version.ClusterVersion,
+		}).Inc()
+		time.Sleep(scrapeTimeout)
+	}
+}
+
+// Function for periodically fetching etcd versioned data size from etcdctl.
+func getVersionedDataSize() {
+	for {
+		// Prepare the etcdctl command to be executed.
+		cmd := exec.Command("/bin/etcdctl", "--endpoints=localhost:2379", "endpoint", "status")
+		env := os.Environ()
+		env = append(env, fmt.Sprintf("ETCDCTL_API=3"))
+		cmd.Env = env
+		cmdOut, _ := cmd.StdoutPipe()
```

Why you are ignoring error?